src/esp/bindings_js/index.js

```diff
@@ -5,17 +5,25 @@ import VRDemo from "./modules/vr_demo";
 import "./bindings.css";
 
 Module.preRun.push(() => {
-  let scene = "skokloster-castle.glb";
+  let config = {};
+  config.scene = "skokloster-castle.glb";
   for (let arg of window.location.search.substr(1).split("&")) {
     let [key, value] = arg.split("=");
-    if (key === "scene" && value) {
-      scene = value;
+    if (key && value) {
+      config[key] = value;
     }
   }
+  const scene = config.scene;
   FS.createPreloadedFile("/", scene, scene, true, false);
   Module.scene = scene;
   const navmesh = scene.substr(0, scene.lastIndexOf(".")) + ".navmesh";
   FS.createPreloadedFile("/", navmesh, navmesh, true, false);
+  if (config.semantic === "mp3d") {
+    const house = scene.substr(0, scene.lastIndexOf(".")) + ".house";
+    FS.createPreloadedFile("/", house, house, true, false);
+    const semantic = scene.substr(0, scene.lastIndexOf(".")) + "_semantic.ply";
+    FS.createPreloadedFile("/", semantic, semantic, true, false);
+  }
```

Comment:
Just checking we're all still on the same page -- once I'm done with the filesystem abstraction APIs (mosra/corrade#39), everything that's added here and in #239 could be moved back to C++, right?

Reply:
I think we are on the same page. I am looking forward to deleting this code once ([mosra/corrade#39](https://github.com/mosra/corrade/pull/39)) lands. I'd much rather just use the C++ filesystem API then have to preload files into the virtual filesystem. This is definitely a temporary hack.